spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/datastore/GcpDatastoreAutoConfiguration.java

```diff
@@ -101,21 +105,28 @@
 	}
 
 	@Bean
-	@ConditionalOnMissingBean
-	public Datastore datastore() {
```

Comment:
Can we keep this been conditional on `DatastoreNamespaceProvider` missing and `Supplier<Datastore>` missing?

Reply:
Based on the highlighting on github I'm not clear on what line you're referring to. 

There is no longer a bean for `Datastore` at all.

Reply:
I'm suggesting to bring it back.

Reply:
I don't understand.. why bring it back if we're not gonna return a real client in there? Isn't this the same exact situation as the default namespace provider?

Reply:
I'm trying to get this to work but I'm running into this issue: 

Because the `Supplier<Datastore>` bean has a `Datastore` param, it causes whatever `Datastore` bean I make to be picked up. The problem is if the user defines his own NamespaceProvider, then I must throw an exception if he tries to request a Datastore instance (since it wouldn't use his namespaceprovider). However that exception gets thrown because the param of the `Supplier<Datastore>` causes that `Datastore` bean to get created. 

I also don't think there's a way to remove the `Datastore` param from the `Supplier<Datastore>` bean because we need to use it if the user defined a `Datastore` himself.

Reply:
@meltsufin

Reply:
@meltsufin I don't follow. The issue isn't the return type of `Supplier<Datastore>`. The issue is that one of the params for that bean is a `Datastore`.